modules/misc/src/materials/ADDensity.C

```diff
@@ -0,0 +1,143 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+#include "ADDensity.h"
+
+registerADMooseObject("MiscApp", ADDensity);
+
+defineADValidParams(
+    ADDensity, ADMaterial, params.addClassDescription("Creates density AD material property");
+    params.addCoupledVar("disp_r", "The r displacement");
+    params.addCoupledVar("disp_x", "The x displacement");
+    params.addCoupledVar("disp_y", "The y displacement");
+    params.addCoupledVar("disp_z", "The z displacement");
+    params.addCoupledVar(
+        "displacements",
+        "The displacements appropriate for the simulation geometry and coordinate system");
+    params.addRequiredParam<Real>("density", "Density"););
+
+template <ComputeStage compute_stage>
+ADDensity<compute_stage>::ADDensity(const InputParameters & parameters)
+  : ADMaterial<compute_stage>(parameters),
+    _is_coupled(true),
+    _disp_r(this->template isCoupled("displacements")
```

Comment:
The `template` keyword is only necessary when you are explicitly specifying a template argument, e.g. with angle brackets `<>`

Reply:
I'm going to push up another patch which adds a few members to `usingMaterialMembers` so that the `this->` syntax is not necessary at all.

Reply:
this needs q_point?

Reply:
@lindsayad adding `using ADKernel<compute_stage>::_q_point; ` to `ADKernel.h` doesn't capture this...

Reply:
whoops, should go in `ADMaterial.h`, but has compiling error...